src/config/field-overrides/program-indicator/AnalyticsPeriodBoundaries.js

```diff
@@ -0,0 +1,182 @@
+import React, { PropTypes } from 'react';
+import { equals, compose } from 'lodash/fp';
+import withState from 'recompose/withState';
+import getContext from 'recompose/getContext';
+import lifecycle from 'recompose/lifecycle';
+
+import { getInstance } from 'd2/lib/d2';
+
+import RaisedButton from 'material-ui/RaisedButton/RaisedButton';
+import FontIcon from 'material-ui/FontIcon/FontIcon';
+
+import DropDown from '../../../forms/form-fields/drop-down';
+import TextField from '../../../forms/form-fields/text-field';
+import PeriodTypeDropDown from '../../../forms/form-fields/period-type-drop-down';
+
+// just wrap the component in a span with some margins as they are
+// inline in this form
+const withStyle = function withStyle(BaseField, styles) {
+    return props => (
+        <span style={styles}>
+            <BaseField {...props}/>
+        </span>
+    );
+};
+
+// boundary type definitions mapped to java enum
+// dhis2-core/dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/AnalyticsPeriodBoundaryType.java#L37-L40
+// as they are not exposed over the API
+function AnalyticsPeriodBoundary (props) {
+    const style = {
+        marginRight: '14px'
+    }
+
+    const getTranslation = props.d2.i18n.getTranslation.bind(props.d2.i18n);
+
+    const DD = withStyle(DropDown, style);
```

Comment:
Not loving these names.

Reply:
Will update with hungarian notation.. ;o

Reply:
7d61b3c